src/Views/DogDetails/DogDetails.js

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useState } from 'react';
+import { useHistory, Link, useParams } from 'react-router-dom';
+import { deleteDog, fetchDogById } from '../../services/fetchdogs';
+
+
+export default function DogDetails({ currentUser }) {
+  const [dog, setDog] = useState('');
+  const [error, setError] = useState('');
+  const [loading, setLoading] = useState('');
```

Comment:
It appears that there is no default value for loading state to be rendering here.